### PR TITLE
Improve Time Model docs

### PR DESCRIPTION
**Pull request by bame-da**
CHANGELOG_BEGIN
CHANGELOG_END

Splitting up #7277.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/app-arch.rst [hunk 1]
```diff
@@ -189,21 +189,22 @@ new endpoint, it will resume normal operation.
...
-- *Ledger time* is reasonably close to real time. To avoid transactions being rejected because the assigned *ledger time* does not match the ledger's system time exactly, DAML Ledgers define a tolerance interval around its system time. The system time is part of the ledger synchronization/consensus protocol, but is not known by the participant node at interpretation time. Transactions with a *ledger time* outside this tolerance interval will be rejected.
-- *Ledger time* respects causal monotonicity: if a transaction ``x`` uses a contract created in another transaction ``y``, transaction ``x``\ s ledger time will be greater than or equal to the ledger time of the referenced transaction ``y``.
+- As part of command interpretation, each transaction is automatically assigned a *Ledger Time* by the participant server.
+- All calls to ``getTime`` within a transaction return the *Ledger Time* assigned to that transaction.
+- *Ledger Time* is chosen (and validated) to respect Causal Monotonicity: The Create action on a contract *c* always precedes all other actions on *c* in Ledger Time.
```
Comment (cocreature) [inline docs/source/app-dev/app-arch.rst:198]:
```suggestion
- *Ledger Time* is chosen (and validated) to respect Causal Monotonicity: The Create action on a contract *c* always precedes (or is equal to) all other actions on *c* in Ledger Time.
```

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: docs/source/app-dev/app-arch.rst (hunk 1 above)
Comment (bame-da) [reply docs/source/app-dev/app-arch.rst:198]: I'm not sure that clarifies things. Plus I'd argue that if "all other actions" is the empty set, the statement is true as well.

Comment (cocreature) [reply docs/source/app-dev/app-arch.rst:198]: The statement before clearly had a `<=` relationship. `precedes` is somewhat ambiguous in whether you force strict inequality or not.

File: docs/source/app-dev/app-arch.rst [hunk 2]
```diff
@@ -189,21 +189,22 @@ new endpoint, it will resume normal operation.
...
+Some commands might take a long time to process, and by the time the resulting transaction is about to be committed to the ledger, it might violate the condition that *Ledger Time* should  be reasonably close to *Record Time* (even when considering the ledger's tolerance interval). To avoid such problems, applications can set the optional parameters :ref:`min_ledger_time_abs <com.daml.ledger.api.v1.Commands.min_ledger_time_abs>` or :ref:`min_ledger_time_rel <com.daml.ledger.api.v1.Commands.min_ledger_time_rel>` that specify (in absolute or relative terms) the minimal *Ledger Time* for the transaction. The ledger will then process the command, but wait with committing the resulting transaction until *Ledger Time* fits within the ledger's tolerance interval.
 
 How is this used in practice?
 
-- Be aware that ``getTime`` is only reasonably close to real time. Avoid DAML workflows that rely on very accurate time measurements or high frequency time changes.
+- Be aware that ``getTime`` is only reasonably close to real time, and not completely monotonic. Avoid DAML workflows that rely on very accurate time measurements or high frequency time changes.
```
Comment (sofiafaro-da) [inline docs/source/app-dev/app-arch.rst:206]: Doesn't `and not completely monotonic` somewhat contradict the earlier `*Ledger Time* is chosen (and validated) to respect Causal Monotonicity: The Create action on a contract *c* always precedes all other actions on *c* in Ledger Time.`?

I guess there's a technical distinction being made but I can't see what it is without some examples. Or maybe "completely monotonic" should be more spelled out?

**Pull request merged**